Manifold/Type.swift

```diff
@@ -14,21 +14,109 @@ public enum Type: Hashable {
 	}
 
 
+	public enum BaseType: Hashable, Printable {
```

Comment:
I also intend these to be defined by type constructors and/or constraints eventually, e.g. `Unit` being the empty product type, and `Bool` being the sum of `true` and `false` constructors.